### PR TITLE
Add Info button to charity selection card.

### LINKED ISSUE #350: Adjust UI for charity card
- [x] Add info button at top to open modal
- [x] Make clicking the card select charity instead.

### PR DESCRIPTION
**Pull request by bharathcs**
Fixes #350 

On large screen:
image

On just med:
image

On small:
image

handle Click / open:

https://user-images.githubusercontent.com/7360964/199061210-5cd07fd1-f627-41bd-b8e7-7ddeba68ba47.mp4

### CODE REVIEW

File: frontend/components/campaigns/campaignCharities/CampaignCharityCard.tsx [hunk 1]
```diff
@@ -24,15 +30,28 @@ const CampaignCharityCard = ({ campaignCharity, onClick }: Props) => {
       </Grid>
 
       <Grid item xs={10}>
-        <Stack paddingTop={1} paddingBottom={1} paddingRight={1}>
-          <Typography variant="h4">{campaignCharity.charity.name}</Typography>
+        <Stack direction="row" spacing={-1} sx={charityGraphInfoStackSx}>
```
Comment (zognin) [inline frontend/components/campaigns/campaignCharities/CampaignCharityCard.tsx:33]:
```suggestion
        <Stack direction="row" sx={charityGraphInfoStackSx}>
```
If we adopt the width approach

File: frontend/styles/components/charities/CampaignCharityCardStyles.ts [hunk 2]
```diff
@@ -7,6 +7,24 @@ export const gridSx: SxProps = {
...
+  paddingRight: 1,
+  flexGrow: 1,
+};
+
+export const charityInfoBoxSx: SxProps = {
+  marginTop: '1px',
+  marginRight: '1px',
+  marginBottom: 'auto',
+  marginLeft: '-1px',
+};
```
Comment (zognin) [inline frontend/styles/components/charities/CampaignCharityCardStyles.ts:26]: If the above is changed, we can remove this

**Review by zognin — APPROVED** [review]
LGTM, pre-approving with some suggestions. There's a related issue in #368 but we can merge this before fixing that.

File: frontend/styles/components/charities/CampaignCharityCardStyles.ts [hunk 3]
```diff
@@ -7,6 +7,24 @@ export const gridSx: SxProps = {
...
 
+export const charityGraphInfoStackSx: SxProps = {
+  width: '100%',
+};
+
+export const charityGraphStackSx: SxProps = {
+  paddingTop: 1,
+  paddingBottom: 1,
+  paddingRight: 1,
+  flexGrow: 1,
```
[the remark this answers is not in the document]
Comment (bharathcs) [reply frontend/styles/components/charities/CampaignCharityCardStyles.ts]: Yep, 1 is the same. I've adopted the width over spacing already!

**Pull request merged**